lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -133,4 +138,55 @@ public function requiresQueryForServerVersion()
     {
         return false;
     }
+
+    /**
+     * Tracks the last insert ID at the current state.
+     *
+     * If this PDO driver is not able to fetch the last insert ID for identity columns
+     * without influencing connection state or transaction state, this is a noop method.
+     *
+     * @internal this method is only supposed to be used in DBAL internals
+     *
+     * @throws \PDOException
+     */
+    public function trackLastInsertId() : void
```

Comment:
The whole design decision behind this method is a bit shady - making subclasses override/disable this behavior looks a bit weird. Would love to see this extracted into some InsertIdStrategyTracking with multiple providers (no-op/identity/...).